pkg/registry/discovery/endpointslice/strategy_test.go

```diff
@@ -0,0 +1,448 @@
+	}{
+		{
+			name:                   "gate enabled, field should be allowed",
+			terminatingGateEnabled: true,
+			eps: &discovery.EndpointSlice{
+				Endpoints: []discovery.Endpoint{
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   utilpointer.BoolPtr(true),
+							Terminating: utilpointer.BoolPtr(false),
+						},
+					},
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   utilpointer.BoolPtr(true),
+							Terminating: utilpointer.BoolPtr(true),
+						},
+					},
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   nil,
+							Terminating: nil,
+						},
+					},
+				},
+			},
+			expectedEPS: &discovery.EndpointSlice{
+				Endpoints: []discovery.Endpoint{
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   utilpointer.BoolPtr(true),
+							Terminating: utilpointer.BoolPtr(false),
+						},
+					},
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   utilpointer.BoolPtr(true),
+							Terminating: utilpointer.BoolPtr(true),
+						},
+					},
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   nil,
+							Terminating: nil,
+						},
+					},
+				},
+			},
+		},
+		{
+			name:                   "gate disabled, field should be set to nil",
+			terminatingGateEnabled: false,
+			eps: &discovery.EndpointSlice{
+				Endpoints: []discovery.Endpoint{
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   utilpointer.BoolPtr(true),
+							Terminating: utilpointer.BoolPtr(false),
+						},
+					},
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   utilpointer.BoolPtr(true),
+							Terminating: utilpointer.BoolPtr(true),
+						},
+					},
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   nil,
+							Terminating: nil,
+						},
+					},
+				},
+			},
+			expectedEPS: &discovery.EndpointSlice{
+				Endpoints: []discovery.Endpoint{
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   nil,
+							Terminating: nil,
+						},
+					},
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   nil,
+							Terminating: nil,
+						},
+					},
+					{
+						Conditions: discovery.EndpointConditions{
+							Accepting:   nil,
+							Terminating: nil,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.EndpointSliceTerminatingCondition, testcase.terminatingGateEnabled)()
+
+			dropDisabledConditionsOnCreate(testcase.eps)
+			if !apiequality.Semantic.DeepEqual(testcase.eps, testcase.expectedEPS) {
+				t.Logf("actual endpointslice: %v", testcase.eps)
+				t.Logf("expected endpointslice: %v", testcase.expectedEPS)
+				t.Errorf("unexpected EndpointSlice")
```

Comment:
This output seems like it might be a bit confusing if it failed. Maybe a more descriptive error message could help.